test/test_datasets.py

```diff
@@ -1502,5 +1470,138 @@ def _create_annotations_file(self, root, name, images, num_captions_per_image):
                 fh.write(f"{image.name}#{idx}\t{caption}\n")
 
 
+class MNISTTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.MNIST
+
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(train=(True, False))
+
+    _MAGIC_DTYPES = {
```

Comment:
Looks like we're only using a subset of these. In particular it seems we're not using any float dtype so maybe we could simplify some stuff, e.g `        if dtype.is_floating_point:` below

Reply:
I'm not opposed to that in general, but our `MNIST` implementation also supports all `dtype`'s:

https://github.com/pytorch/vision/blob/19ad0bbc5e26504a501b9be3f0345381d6ba1efc/torchvision/datasets/mnist.py#L471-L478

So in theory one could add a new `MNIST` variant that uses float and it works out of the box for the actual implementation but the tests need to be adapted. Thus, I've opted to include the complete list here.

Reply:
I think we should only introduce code when we need it. If we were to run coverage on the test, we would find out (I think) that these bits below are never run:

```py
        if dtype.is_floating_point:

            def data_fn(size, dtype):
                return torch.rand(size, dtype=dtype)
```

It's one thing to future-proof a design (and I'm all for it), but it's another thing to introduce code that's never being run just because we might need it in the future. In addition that code, might actually be buggy but we wouldn't even know since it's not being run